generator/flat/templates/record.go

```diff
@@ -6,9 +6,17 @@ import (
 	"github.com/actgardner/gogen-avro/vm/types"
 	"github.com/actgardner/gogen-avro/vm"
 	"github.com/actgardner/gogen-avro/compiler"
+	"github.com/actgardner/gogen-avro/schema/canonical"
 )
 
-{{ if ne .Doc "" }}// {{ .Doc}}{{ end }}  
+var {{ .Name }}UID []byte
```

Comment:
This should be a `const` we compute when we generate the code, rather than re-computing it at package initialization time.

Reply:
I will have to code generate the byte array as a const. I'm not very familiar with how the code generation work in go and the doc resources for it are very limited. I will try to figure it out otherwise I will ask for some help from you.

Reply:
The simplest thing to do would be to add a method to the record definition type, and then call it in the template like `{{ .AvroUID }}`

Reply:
I was working on it and did exactly that. The issue I'm facing is that I need the parser to calculate the canonical form and generate the UID. it create a cyclic import issue schema-->parser-->schema. Any idea on how to resolve that?

Reply:
I pushed a commit to master that adds the hex-encoded CRC64 for the canonical schema as a constant to generated record types, it took a little bit more finagling than expected.

Reply:
I see, is it ok if I change the type to a slice of byte? If it's a hex string extra conversion is needed for each read and write.

Reply:
I merged your commit and made the changes to generate the fingerprint in []byte